src/Sarif.WorkItems/SarifWorkItemModel.cs

```diff
@@ -43,11 +43,9 @@ public SarifWorkItemModel(SarifLog sarifLog, SarifWorkItemContext context = null
                 Text = JsonConvert.SerializeObject(sarifLog),
             };
 
-            // TODO: Provide a useful SARIF-derived issue title
-            //
-            //       https://github.com/microsoft/sarif-sdk/issues/1755
-            //
-            this.Title = $"Default {nameof(this.Title)}";
+            string title = sarifLog.Runs?[0]?.CreateWorkItemTitle();
+
+            this.Title = title ??  "[ERROR GENERATING TITLE]";
```

Comment:
>[ERROR GENERATING TITLE] [](start = 36, length = 24)

1. When Alison is done with the logging, we will record any substantive problems trying to produce the work item. Alternately, we could raise an exception and refuse to complete the work. Alternately, we could improve on this error and keep processing. Thoughts?

Reply:
In a scenario where there are thousands of bugs to create and there is an error in generating the title, would the users be ok having a tons of incorrect/modified titles? I'm trying to see it from a customers' perspective.

---
In reply to: [392391848](https://github.com/microsoft/sarif-sdk/pull/1813#discussion_r392391848) [](ancestors = 392391848)

Reply:
Nope, this would be a bad customer experience. Do you have a suggestion?

---
In reply to: [392450784](https://github.com/microsoft/sarif-sdk/pull/1813#discussion_r392450784) [](ancestors = 392450784,392391848)

Reply:
I guess I would fail fast and provide lots of logging for developers to debug and/or customers to fix up their input if that's the case. 

---
In reply to: [392490164](https://github.com/microsoft/sarif-sdk/pull/1813#discussion_r392490164) [](ancestors = 392490164,392450784,392391848)

Reply:
I tend to agree. The implications of this approach are that a corner case issue with title construction could complete torpedo a bug filing session for a specific tool. By failing entirely, we put pressure (which could be useful) on acquiring a tool fix as soon as possible. Tool updates themselves come with complications (as in many cases, a spot fix for a detail issue like this might not be applied to the current version. Instead, the upgrade path may imply receiving a minor/major update with analysis delta). All of this makes a point I repeat all the time: our systems need to be designed for minimizing friction for tool servicing changes. It's interesting to see how this local decision in behavior needs to connect to a pretty high-level design philosophy. :) In any case, I will make this fail-fast for now (in the next clean-up change).

---
In reply to: [392501325](https://github.com/microsoft/sarif-sdk/pull/1813#discussion_r392501325) [](ancestors = 392501325,392490164,392450784,392391848)